skypy/clusters/sampler.py

```diff
@@ -0,0 +1,101 @@
+'''Halo mass sampler.
+This code samples cluster halos from the mass function (Despali+ 16).
+
+'''
+import numpy as np
+import colossus as colossus
+import astropy as astropy
+from astropy.cosmology import WMAP9
+from colossus.cosmology import cosmology
+from colossus.lss import mass_function
+from scipy import integrate
+import scipy
+
+
+def selection(z):
+    """
+        eRosita selection function from Pillepich et al. 2012.
+        Including mass cut in the low mass end.
+        func(x) is a fitting function based on the data point presented in Pillepich et al. 2012.
+    Parameters
+    -----------
+    z: float
+     Redshift
+            
+    Returns
+    --------
+    fit_M: array_like
+     minimum halo mass (mdef: 500c, unit: Msun) that can be detected by eRosita
+
+    """
+    
+    def func(x):
+        a = 9.26958708e+04
+        b = -4.55726030e+00
+        c = 1.52774801e+00
+        d = -1.85377292e+05
+        return a* (1+scipy.special.erf(((x)- b ) / c )  ) + d
+    
+    if z <= 0.1:  ##apply mass cut: remove halo with halo mass M500c<1e+13
+        fit_M = 1e+13
+    else:
+        p = func(z)
+        fit_M = 10**(p)
+    return fit_M  ##unit: Msun
+
+def mass_sampler(z, size, mdef = '500c', m_min = 1e+12, m_max = 1e+16, select = False, newcosmo_astropy = WMAP9, sigma8 = 0.8200, ns = 0.9608):
```

Comment:
```suggestion
def colossus_mass_sampler(redshift, model, mdef, m_min, m_max, cosmology, sigma8, ns, size=None):
```
- Rename the function for clarity
- Rename `z` to `redshift` for clarity
- `size=None` should have the same default value as for `np.random.uniform`
- Add `resolution=1000` as a default argument that the user can adjust rather than hardcoding
- No other arguments should have default values
- Make the function more general by taking the colossus `model` as an argument as well
- Remove the selection effect from this sampler. It is too specific (works for eRosita only) and we can calculate it in a separate step in the pipeline since it effectively just determines `m_min`
- Rename `newcosmo_astropy` to `cosmology` since the SkyPy Pipeline class recognises this name
- Update the docstrings to reflect these changes